runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -42,7 +46,22 @@
 @Slf4j
 public class Table<K extends Message, V extends Message, M extends Message> {
 
-    private CorfuTable<K, CorfuRecord<V, M>> corfuTable;
+    // Accessor/Mutator threads can interleave in a way that create a deadlock because they can create a
+    // circular dependency between the VersionLockedObject(VLO) lock and the common forkjoin thread pool. In order
+    // to break the dependency, parallel stream operations have to execute on a separate pool that applications
+    // cant use. For example, if there are 4 accessor threads all using the common forkjoin pool, one of the threads
+    // can acquire the VLO lock and cause the other 3 threads to wait, but after acquiring the VLO lock, the thread
+    // gets block on parallel stream, because the pool is exhausted with threads that are trying to acquire the VLO
+    // look, which creates a circular dependency. In other words, a deadlock.
+
+    protected static final ForkJoinPool pool = new ForkJoinPool(Math.max(Runtime.getRuntime().availableProcessors() / 2, 1),
+            pool -> {
+                final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
+                worker.setName("Table-Forkjoin-pool-" + worker.getPoolIndex());
+                return worker;
+            }, null, true);
```

Comment:
Can you change this to a fixed thread pool and make the number of workers configurable via the Runtime options?

The problem with Forkjoin pool is that it can fork more threads than specified if some of the work is blocking. The scan and filter shouldn't have blocking code. Having a fixed thread pool would make this issue easier to observe in a profiler.